[Text] Design guidelines (#303)
primer/brand: apps/docs/content/components/Text/index.mdx

---
title: Text

> description: Use the text component to render body content

Reviewer: The term "body content" generally encompasses paragraphs and headings (among other textual components) in the main text of a document. Should we perhaps be more specific in this description, as I believe the text component is not used for headings?

---

import CustomVideoPlayer from '../../../src/components/custom-video-player'
import ComponentLayout from '../../../src/layouts/component-layout'
export default ComponentLayout

> ![An image displaying each text from size 100 to 700.](https://github.com/primer/brand/assets/912236/67d89861-7a1e-4df9-a801-01d11f4dfb17)

Reviewer 2: ```suggestion
![The text component has size ranges from 100 to 700 with 100 being the smallest, and 700 being the largest.](https://github.com/primer/brand/assets/912236/67d89861-7a1e-4df9-a801-01d11f4dfb17)
```

I think it might make more sense to just describe what the image is showing, or not have alt text at all

## Usage

> Use the text component to provide content and a clear hierarchy structure. The text component renders an HTML `<span>` tag without any default styling. Use `<p>` for any text that is meant to be a paragraph of text. Use multiple text components or `<p> `if needed to structure your content when you have distinct blocks of text that represent individual paragraphs. This helps with readability and makes it easier for search engines and assistive technologies to understand the content.

Reviewer 2: ```suggestion
Use the text component to provide content and a clear hierarchy structure. The text component renders an HTML `<span>` tag without any default styling. Use `<p>` for any text that is meant to be a paragraph of text. Use multiple text components to represent paragraphs as distinct blocks of text. This helps with readability and makes it easier for search engines and assistive technologies to understand the content.
```

Author: love this, thank you!

## Options

### Variant

The text component provides two variant options: `default` and `muted`. The `muted` variant is used to de-emphasize text or to provide visual contrast when paired with a title.

<DoDontContainer>
  <Do>
    <img src="https://github.com/primer/brand/assets/912236/b7fa8c0d-0877-4edd-a8b3-4eea2e16fe26" />
    <Caption>
      Use muted text to de-emphasize text and provide additional context.
    </Caption>
  </Do>
  <Dont>
    <img src="https://github.com/primer/brand/assets/912236/01f1e3a8-5058-4a73-a69f-6f424382fb63" />
    <Caption>
      Don't use standalone muted text. Muted text should always be paired with
      emphasized text or a title.
    </Caption>
  </Dont>
</DoDontContainer>

### Size

> The text size ranges from 100 to 700, being 400 the default size. This scale is built to provide hierarchy for all types of scenarios.

Reviewer 2: ```suggestion
The text size ranges from 100 to 700, with 400 as the default size. This scale is built to provide hierarchy for all types of scenarios.
```

<DoDontContainer>
  <Do>
    <img src="https://github.com/primer/brand/assets/912236/e029b053-eab8-47a7-bd49-b6ce49ac8bdf" />

>     <Caption>
>       Use a smaller size to provide visual hierarchy when the text is paired
>       with a title. Use as many text as needed to represent different blocks of
>       content.
>     </Caption>

Reviewer 2: ```suggestion
    <Caption>
      Use a smaller size to provide visual hierarchy when the text is paired
      with a title. Use as many text components as needed to represent different blocks of
      content.
    </Caption>
```
